package/kedro_viz/server.py

```diff
@@ -398,70 +414,114 @@ def format_pipeline_data(
                 "full_name": getattr(node, "_func_name", str(node)),
                 "tags": sorted(node.tags),
                 "pipelines": [pipeline_key],
+                "modular_pipelines": _expand_namespaces(node.namespace),
             }
             nodes_list.append(nodes[task_id])
         else:
             nodes[task_id]["pipelines"].append(pipeline_key)
 
         for data_set in node.inputs:
-            namespace = data_set.split("@")[0]
-            namespace_to_layer[namespace] = dataset_to_layer.get(data_set)
-            namespace_id = _hash(namespace)
+            dataset_namespace = data_set.split("@")[0]
+            if node.namespace:
+                namespace_to_modular_pipelines[dataset_namespace] = _expand_namespaces(
+                    node.namespace
+                )
+            dataset_namespace_to_layer[dataset_namespace] = dataset_to_layer.get(
+                data_set
+            )
+            namespace_id = _hash(dataset_namespace)
             edge = {"source": namespace_id, "target": task_id}
             if edge not in edges_list:
                 edges_list.append(edge)
-            namespace_tags[namespace].update(node.tags)
+            dataset_namespace_tags[dataset_namespace].update(node.tags)
             node_dependencies[namespace_id].add(task_id)
 
             # if it is a parameter, add it to the node's data
-            if _is_namespace_param(namespace):
-                if "parameters" not in _JSON_NODES[task_id]:
-                    _JSON_NODES[task_id]["parameters"] = {}
-
-                if namespace == "parameters":
-                    _JSON_NODES[task_id]["parameters"] = _get_dataset_data_params(
-                        namespace
-                    ).load()
-                else:
-                    parameter_name = namespace.replace("params:", "")
-                    parameter_value = _get_dataset_data_params(namespace).load()
-                    _JSON_NODES[task_id]["parameters"][parameter_name] = parameter_value
+            if _is_namespace_param(dataset_namespace):
+                _add_parameter_data_to_node(dataset_namespace, task_id)
 
         for data_set in node.outputs:
-            namespace = data_set.split("@")[0]
-            namespace_to_layer[namespace] = dataset_to_layer.get(data_set)
-            namespace_id = _hash(namespace)
+            dataset_namespace = data_set.split("@")[0]
+            if node.namespace:
+                namespace_to_modular_pipelines[dataset_namespace] = _expand_namespaces(
```

Comment:
The belonging to modular pipelines of a dataset is not determined by the nodes using it, but only by it's name. This one assigns more modular pipelines for a dataset and will cause a lot of confusion when expanding / collapsing pipelines. E.g. what will happen if I collapse one of the modular pipelines this belongs to?

The hierarchical relationship is exclusive unlike tags, i.e. a node / dataset can belong to only one hierarchy of namespaces.